2010-05-31  Michael Tautschnig  <mt@debian.org>

	* fai-chboot, fai-chboot.8: Also accept mac addresses as host specifications
		(thanks Torge Gipp <post@torge-gipp.de>)
Index: trunk/bin/fai-chboot
===================================================================
--- trunk.orig/bin/fai-chboot
+++ trunk/bin/fai-chboot	
@@ -55,6 +55,13 @@
     return ip2hex($host);
   }
   return ('no IP','default') if ($host =~ /^default/);
+  # mac address using : or - as separators
+  if ($host =~ /^([[:xdigit:]]{2}[:-]){5}[[:xdigit:]]{2}$/) { # Pattern-Matching
+	  $host = lc $host; # Lowercase for pxe-file
+	  my @nums = split /[:-]/, $host;
+	  $mac = "01-$nums[0]-$nums[1]-$nums[2]-$nums[3]-$nums[4]-$nums[5]"; #
+	  return ($host,$mac);
+  }
 
   my $h = gethost($host);
   die "$0: unknown host: $host\n" unless $h;
@@ -252,7 +259,7 @@
   }
   open (FILE,"> $pxedir/$hex") or warn "$0 $@ $!";
   print FILE << "EOM";
-# generated by fai-chboot for host $host with IP $ipadr
+# generated by fai-chboot for host $host with address $ipadr
 default fai-generated
 
 label fai-generated
@@ -301,7 +308,7 @@
   my ($srcfile, $pxedir, $desthex, $desthost, $ipadr) = @_;
   open (SOURCE, "$srcfile");
   open (DEST, ">$pxedir/$desthex") || die "Can't opendir $pxedir: $!";
-  print DEST "# template generated by fai-chboot for host $desthost with IP $ipadr from source $srcfile\n";
+  print DEST "# template generated by fai-chboot for host $desthost with address $ipadr from source $srcfile\n";
   while (<SOURCE>) {
     print DEST $_;
   }
Index: trunk/man/fai-chboot.8
===================================================================
--- trunk.orig/man/fai-chboot.8
+++ trunk/man/fai-chboot.8	
@@ -40,7 +40,8 @@
 additional kernel parameters and define some FAI specific flags. For
 each host one configuration file is created. The file name of the
 configuration file is its IP address in upper case hexadecimal. HOST
-can be a host name, an IP address or the keyword "default".
+can be a host name, an IP address, a MAC address (hexadecimal words
+separated by ":" or "\-"), or the keyword "default".
 
 .SH OPTIONS
 .TP
